### ISSUE TITLE
could you share training parameters?

### ISSUE THREAD

**Issue by phoenixkillerli** [tancik/StegaStamp#17]
I try to train model, bug could not get the same performance like the released model.
Could you share training parameters or told me how to adjust parameters.

Thanks.

**Comment by tancik** [comment]
The released model was trained using `bash scripts/base.sh EXP_NAME`.
What differences are you finding in performance?

**Comment by phoenixkillerli** [comment]
I used MIRFLICKR-25000 image collection to train model. and then test encode and decode some images. All are failed. But use the released model, all are succeed.
I will try again and find problem.
Thanks.

**Comment by fujiaixin09** [comment]
> I used MIRFLICKR-25000 image collection to train model. and then test encode and decode some images. All are failed. But use the released model, all are succeed.
> I will try again and find problem.
> Thanks.

Hello! Have you solved this problem? I have the same problem

**Comment by EvgenGunya** [comment]
Hello Guys, have anybody solver this problem?
training with original parameters does not work at all in my case.

**Comment by fujiaixin09** [comment]
> Hello Guys, have anybody solver this problem?
> training with original parameters does not work at all in my case.

No, I use the code and parameters published by the author, but the accuracy of the training can not meet the requirements of the author. I don't know why.

**Comment by EvgenGunya** [comment]
fujiaixin09

> > Hello Guys, have anybody solver this problem?
> > training with original parameters does not work at all in my case.
> 
> No, I use the code and parameters published by the author, but the accuracy of the training can not meet the requirements of the author. I don't know why.

Hello fujiaixin09,
Thank you for your answer! 
But when you said that  accuracy couldn't meet. Is it mean that sometimes it works (model can decode)? In my case it is just nothing, it looks like CNN was not trained at all.

**Comment by fujiaixin09** [comment]
No, it can't decode. The bit accuracy of my training model can only reach more than 80.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tancik/StegaStamp"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年5月23日(星期天) 晚上6:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tancik/StegaStamp] could you share training parameters? (#17)





 
fujiaixin09
   
Hello Guys, have anybody solver this problem?
 training with original parameters does not work at all in my case.
  
No, I use the code and parameters published by the author, but the accuracy of the training can not meet the requirements of the author. I don't know why.
  
Hello fujiaixin09,
 Thank you for your answer!
 But when you said that  accuracy couldn't meet. Is it mean that sometimes it works (model can decode)? In my case it is just nothing, it looks like CNN was not trained at all.

**Comment by EvgenGunya** [comment]
> The released model was trained using `bash scripts/base.sh EXP_NAME`.
> What differences are you finding in performance?

Hello Tancik, 
Thank you for your repository  it is really cool!
Can you suggest us anything about training, it looks nobody can train model that will be able to decode watermarked image. But at the same time I was able to run your model for decoding and it works fine.

**Comment by tancik** [comment]
Can you confirm the following,

_ No modification of the code were makes
_ The parameters in `scripts/base.sh` are being used
_ The MIRFLICKR data is  being used for training
_ tf1.3 is being used

Additionally, how many iterations do you train for and how long does this training take (in hours).
Can you post a screenshot of the tensorboard after training.

**Comment by EvgenGunya** [comment]
> Can you confirm the following,
> 
> _ No modification of the code were makes
> _ The parameters in `scripts/base.sh` are being used
> _ The MIRFLICKR data is being used for training
> _ tf1.3 is being used
> 
> Additionally, how many iterations do you train for and how long does this training take (in hours).
> Can you post a screenshot of the tensorboard after training.

Hello @tancik,
Thank you for fast answer,

_ No modification of the code were makes
 - no modification just git clone

_ The parameters in scripts/base.sh are being used
 - yes, I have run as described in https://github.com/tancik/StegaStamp/blob/master/README.md

_ The MIRFLICKR data is being used for training
 - yes MIRFLICKR it has 25000 images

_ tf1.3 is being used
 - if I use tf1.3 - it gives 
error: cannot import 'keras'. And after fixing this, new errors appeared. I think original code has been written on the other tf
 - if I use tf1.13 as it described in README it works but after training it is not recognised anything.

how many iterations do you train
 - 140 000 as should be

how long does this training take (in hours).
 - I think +- 12 hours on gtx 1080 ti

screenshot in attachment.

Kind regards
![Screenshot from 2021-06-03 05-16-42](https://user-images.githubusercontent.com/5990818/120576476-7c1f5c80-c42b-11eb-9018-6329d3edad0b.png)

**Comment by tancik** [comment]
Hmm, interesting. Can you also provide a screenshot of the remaining scalar logs and screenshot of the image logs.

**Comment by EvgenGunya** [comment]
Yes sure.
![Screenshot from 2021-06-04 20-08-19](https://user-images.githubusercontent.com/5990818/120838962-1341fc80-c571-11eb-881c-bba3b37eb1cc.png)
![Screenshot from 2021-06-04 20-08-29](https://user-images.githubusercontent.com/5990818/120838965-13da9300-c571-11eb-8182-8c221bf2652f.png)
![Screenshot from 2021-06-04 20-08-35](https://user-images.githubusercontent.com/5990818/120838968-14732980-c571-11eb-98c3-8a0f510f1a2d.png)
![Screenshot from 2021-06-04 20-08-57](https://user-images.githubusercontent.com/5990818/120838973-150bc000-c571-11eb-9b79-45d11fb7b5ff.png)
![Screenshot from 2021-06-04 20-09-00](https://user-images.githubusercontent.com/5990818/120838978-15a45680-c571-11eb-94b4-8dd0a0c70163.png)
![Screenshot from 2021-06-04 20-10-32](https://user-images.githubusercontent.com/5990818/120838979-163ced00-c571-11eb-96b5-cf90871978b6.png)
![Screenshot from 2021-06-04 20-10-46](https://user-images.githubusercontent.com/5990818/120838981-16d58380-c571-11eb-9f97-9fc2379122df.png)

**Comment by tancik** [comment]
It is odd that the input images are black. It doesn't seem to be using the dataset images.

**Comment by EvgenGunya** [comment]
hmm. yes, I see.

**Comment by Github-Shaw** [comment]
hello, I ran into the same problem。
could you tell me how to solve it，pls？
@EvgenGunya

**Comment by Github-Shaw** [comment]
> hmm. yes, I see.
hello, I ran into the same problem。
could you tell me how to solve it，pls？

**Comment by Github-Shaw** [comment]
> hmm. yes, I see.


sdasd

**Comment by Github-Shaw** [comment]
> hmm. yes, I see.


hello, I ran into the same problem。
could you tell me how to solve it，pls？

**Comment by EvgenGunya** [comment]
My problem was path to the image folder.

**Comment by Github-Shaw** [comment]
> My problem was path to the image folder.

thx
I modified  the variable  TRAIN_PATH of  train.py, but useless
Do I need to modify other variables?
@EvgenGunya

**Comment by prasenjitennoventure** [comment]
@phoenixkillerli  or anyone I am facing the similar challenges. Can you or anyone please share the actual released model
as the released model is not accessible and the recent shared model via google drive link does not contain proper checkpointing file. 

Thanks in advance.
